path_patching.py

```diff
@@ -847,6 +851,109 @@ def act_patch(
             results_dict[node_name].append(act_patch_single(patching_nodes=node))
             progress_bar.update(1)
     progress_bar.close()
+    for node_name, node_shape_dict in patching_nodes.shape_values.items():
+        if verbose: print(f"results[{node_name!r}].shape = ({', '.join(f'{s}={v}' for s, v in node_shape_dict.items())})")
+    return {
+        node_name: t.tensor(results).reshape(list(patching_nodes.shape_values[node_name].values())) if isinstance(results[0], float) else results
+        for node_name, results in results_dict.items()
+    }
+
+
+def train_probe_at_layer_pos(model, act_folder, labels, layer, pos=-1, component='hook_resid_post', with_scaler=True, max_iter=100):
+
+    # Get activations
+    with open(os.path.join(act_folder, f"{component}_pos_{pos}_activations.pkl"), "rb") as f:
```

Comment:
Should we add the model to the filename to avoid data being overwritten by mistake?